### PR TITLE
style: introduce code formatting

### PR DESCRIPTION
**Pull request by DavidSanwald**
There were some inconsistencies with code formatting, that sometimes resulted in changed lines in commits, even though there was no actual change in the code.

Prettier is the black of the JavaScript world.
It does not offer a lot of configuration options on purpose and I chose everything standard.
If there are different preferences, we can change, it's more important to have a consistent, deterministic way to format the code.

I added husky and lint-staged so every staged file will be formatted before commit. This  helps with consistency and prevents missleading changed lines in commits.

tldr:
- code formatter
- hooks to format staged files on commit
- initial formatting

### CODE REVIEW

**Comment by nan-wang** [comment]
This is a great Initiative!
